jerry-core/ecma/operations/ecma-regexp-object.c

```diff
@@ -541,725 +625,950 @@ ecma_regexp_match (ecma_regexp_ctx_t *re_ctx_p, /**< RegExp matcher context */
   }
 #endif /* JERRY_STACK_LIMIT != 0 */
 
+  const lit_utf8_byte_t *str_start_p = str_curr_p;
+  const uint8_t *next_alternative_p = NULL;
+
   while (true)
   {
-    re_opcode_t op = re_get_opcode (&bc_p);
+    const re_opcode_t op = re_get_opcode (&bc_p);
 
     switch (op)
     {
-      case RE_OP_MATCH:
+      case RE_OP_EOF:
+      {
+        re_ctx_p->captures_p[RE_GLOBAL_CAPTURE].end_p = str_curr_p;
+        /* FALLTHRU */
+      }
+      case RE_OP_ASSERT_END:
+      case RE_OP_ITERATOR_END:
       {
-        JERRY_TRACE_MSG ("Execute RE_OP_MATCH: match\n");
         return str_curr_p;
       }
-      case RE_OP_CHAR:
+      case RE_OP_ALTERNATIVE_START:
       {
-        if (str_curr_p >= re_ctx_p->input_end_p)
+        const uint32_t offset = re_get_value (&bc_p);
+        next_alternative_p = bc_p + offset;
+        continue;
+      }
+      case RE_OP_ALTERNATIVE_NEXT:
+      {
+        while (true)
         {
-          return NULL; /* fail */
+          const uint32_t offset = re_get_value (&bc_p);
+          bc_p += offset;
+
+          if (*bc_p != RE_OP_ALTERNATIVE_NEXT)
+          {
+            break;
+          }
+
+          bc_p++;
         }
 
-        const bool is_ignorecase = re_ctx_p->flags & RE_FLAG_IGNORE_CASE;
-        lit_code_point_t ch1 = re_get_char (&bc_p); /* Already canonicalized. */
-        lit_code_point_t ch2 = lit_cesu8_read_next (&str_curr_p);
+        continue;
+      }
+      case RE_OP_NO_ALTERNATIVE:
+      {
+        return NULL;
+      }
+      case RE_OP_CAPTURING_GROUP_START:
+      {
+        const uint32_t group_idx = re_get_value (&bc_p);
+        ecma_regexp_capture_t *const group_p = re_ctx_p->captures_p + group_idx;
+        group_p->subcapture_count = re_get_value (&bc_p);
+
+        const lit_utf8_byte_t *const saved_begin_p = group_p->begin_p;
+        const lit_utf8_byte_t *const saved_end_p = group_p->end_p;
+        const uint32_t saved_iterator = group_p->iterator;
 
-#if ENABLED (JERRY_ES2015)
-        if (re_ctx_p->flags & RE_FLAG_UNICODE
-            && lit_is_code_point_utf16_high_surrogate (ch2)
-            && str_curr_p < re_ctx_p->input_end_p)
+        const uint32_t qmin = re_get_value (&bc_p);
+        group_p->end_p = NULL;
+
+        /* If zero iterations are allowed, then execute the end opcode which will handle further iterations,
+         * otherwise run the 1st iteration immediately by executing group bytecode. */
+        if (qmin == 0)
```

Comment:
Is this likely? Can it be handled at compile time in some way?

Reply:
The likelihood would be roughly 50/50 in my opinion, it could be handled compile time, however that would increase code size, and performance impact would be negligible.